clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/provider/agent/OpenstackServerGroupCachingAgent.groovy

```diff
@@ -63,31 +71,50 @@ class OpenstackServerGroupCachingAgent extends AbstractOpenstackCachingAgent {
   ] as Set)
 
   final ObjectMapper objectMapper
+  final OnDemandMetricsSupport metricsSupport
+  final String agentType = "${account.name}/${region}/${OpenstackServerGroupCachingAgent.simpleName}"
+  final String onDemandAgentType = "${agentType}-OnDemand"
 
-  OpenstackServerGroupCachingAgent(OpenstackNamedAccountCredentials account, String region,
-                                   final ObjectMapper objectMapper) {
+  OpenstackServerGroupCachingAgent(final OpenstackNamedAccountCredentials account, final String region,
+                                   final ObjectMapper objectMapper, final Registry registry) {
     super(account, region)
     this.objectMapper = objectMapper
-  }
-
-  @Override
-  String getAgentType() {
-    "${account.name}/${region}/${OpenstackServerGroupCachingAgent.simpleName}"
-  }
-
-  @Override
-  Collection<AgentDataType> getProvidedDataTypes() {
-    types
+    this.metricsSupport = new OnDemandMetricsSupport(registry, this, "${ID}:${ServerGroup}")
   }
 
   @Override
   CacheResult loadData(ProviderCache providerCache) {
     CacheResultBuilder cacheResultBuilder = new CacheResultBuilder(startTime: Long.MAX_VALUE)
```

Comment:
I don't think startTime of Long.MAX_VALUE makes sense - as it is, cacheData start time will always be less than cacheResultBuilder start time.